python/xorbits/core/data.py

```diff
@@ -191,10 +191,13 @@ def __getattr__(self, item):
         return MemberProxy.getattr(self, item)
 
     def __setattr__(self, key, value):
-        if key in self.__fields:
-            object.__setattr__(self, key, value)
-        else:
-            self.data.__setattr__(key, value)
+        try:
+            if key in self.__fields:
+                object.__setattr__(self, key, value)
+            else:
+                self.data.__setattr__(key, value)
+        except AttributeError:
+            self.__setitem__(key, value)
```

Comment:
Maybe we can do as follows:

```
if key in self.dtypes:
    self.__setitem__(key, value)
else:
    # logic before
```

Or assume we set a new column, pandas seems not actually do that. Example:

```
In [55]: df = pd.DataFrame(np.random.rand(4, 3), columns=list('abc'))

In [56]: df
Out[56]: 
          a         b         c
0  0.434708  0.214688  0.219377
1  0.734574  0.303740  0.150693
2  0.142611  0.422218  0.026213
3  0.454294  0.795692  0.443908

In [57]: df.d = [1, 2, 3, 4]
<ipython-input-57-57215c2c3f93>:1: UserWarning: Pandas doesn't allow columns to be created via a new attribute name - see https://pandas.pydata.org/pandas-docs/stable/indexing.html#attribute-access
  df.d = [1, 2, 3, 4]

In [58]: df
Out[58]: 
          a         b         c
0  0.434708  0.214688  0.219377
1  0.734574  0.303740  0.150693
2  0.142611  0.422218  0.026213
3  0.454294  0.795692  0.443908

In [59]: df.b = [1, 2, 3, 4]

In [60]: df
Out[60]: 
          a  b         c
0  0.434708  1  0.219377
1  0.734574  2  0.150693
2  0.142611  3  0.026213
3  0.454294  4  0.443908
```